### PR TITLE
fix: properly skip/cleanup controlplane configs for workers

### PR DESCRIPTION
**Pull request by smira**
This bug is pretty cosmetic, but it shows up as a wrong check when performing worker upgrade - Talos pretends it checks e.g. kube-apiserver version which doesn't make sense for workers.

There were two bugs in the code:

* check for machine type was done against `TypeWorker`, while `MachineType` resource is initially created as `TypeUnknown`
* the cleanup code was not implemented

As I touched the code, I updated controller and tests to use modern conventions.

### CODE REVIEW

File: internal/app/machined/pkg/controllers/k8s/control_plane.go [hunk 1]
```diff
@@ -121,10 +119,8 @@ func (ctrl *ControlPlaneController) Run(ctx context.Context, r controller.Runtim
 			return fmt.Errorf("error getting machine type: %w", err)
 		}
 
-		machineType := machineTypeRes.(*config.MachineType).MachineType()
-
-		if machineType == machine.TypeWorker {
-			if err = ctrl.teardownAll(ctx, r, logger); err != nil {
+		if !machineType.MachineType().IsControlPlane() {
+			if err = ctrl.teardownAll(ctx, r); err != nil {
```
Comment (frezbo) [inline internal/app/machined/pkg/controllers/k8s/control_plane.go:123]: why would we need to delete the resources if it's not controlplane, it shouldn't be there in the first place, right?

**Review by frezbo — APPROVED** [review]

File: internal/app/machined/pkg/controllers/k8s/control_plane.go (hunk 1 above)
Comment (smira) [reply internal/app/machined/pkg/controllers/k8s/control_plane.go:123]: I tried to explain it above, the previous check got into the state when resource was created for worker nodes while `MachineType` was in `TypeUnknown`. This code is more for completeness, once we support changing `MachineType` on the fly one day.

Comment (frezbo) [reply internal/app/machined/pkg/controllers/k8s/control_plane.go:123]: that makes sense. cool

**Pull request merged**